packages/block-editor/src/store/reducer.js

```diff
@@ -1379,11 +1379,27 @@ export function blocksMode( state = {}, action ) {
  */
 export function insertionPoint( state = null, action ) {
 	switch ( action.type ) {
-		case 'SHOW_INSERTION_POINT':
+		case 'SET_INSERTION_POINT': {
 			const { rootClientId, index } = action;
-			return { rootClientId, index };
+			return state
+				? { ...state, rootClientId, index }
+				: { rootClientId, index };
+		}
+
+		case 'SHOW_INSERTION_POINT': {
+			const { rootClientId, index } = action;
+			return { rootClientId, index, isVisible: true };
+		}
 
 		case 'HIDE_INSERTION_POINT':
+			return state ? { ...state, isVisible: false } : state;
```

Comment:
I'm not sure there's any advantage to retaining the rootClientId and index using `...state`, since this is always overwritten by `SHOW_INSERTION_POINT` or `SET_INSERTION_POINT`.

Potentially this could just return `null`.

Reply:
It simplifies things quite a bit to split it into two reducers:
- `insertionPoint` which has the shape `{ rootClientId, index }` and a default value of `{}`. This is set by `SHOW_INSERTION_POINT` and `SET_INSERTION_POINT` and reset by a bunch of other actions including `HIDE_INSERTION_POINT`.
- `insertionPointVisibility` which is a boolean. This is set to `true` by `SHOW_INSERTION_POINT` and `false` by `HIDE_INSERTION_POINT`.

Reply:
Turns out HIDE_INSERTION_POINT should retain the state as per - https://github.com/WordPress/gutenberg/pull/26443#issuecomment-724435982. The component that calls `showInsertionPoint` must be reusing the existing rootClientId and index.

Splitting into two reducers is a bit friendlier anyway, so a worthwhile change.